### PR TITLE
[bitnami/postgresql] feat: :sparkles: Add ServiceBinding-compatible secrets

### PR DESCRIPTION
**Pull request by javsalgar**
### Description of the change

THIS FEATURE IS EXPERIMENTAL

This PR adds a `serviceBindings.enabled` flag that, apart from the regular secret, it will create extra ones to be integrated with the [ServiceBindings](https://servicebinding.io/). In order to not make it very breaking, the flag is disabled by default and also it will not replace the existing secrets. 

It will create two extra secrets: one for the Postgres user and one for the custom user. It can be simply integrated with a ServiceBinding object out of the box. 

### Applicable issues

- fixes #

### Checklist

- [ ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [ ] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [ ] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [ ] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/postgresql/values.yaml [hunk 1]
```diff
@@ -1158,6 +1164,12 @@ volumePermissions:
     runAsUser: 0
 
 ## @section Other Parameters
+##
+
+## @param serviceBindings.enabled Create secret for service binding (Experimental)
```
Comment (mpermar) [inline bitnami/postgresql/values.yaml:1169]:
```suggestion
## @param serviceBindings.enabled Create secret for service binding (Experimental)
## Ref: https://servicebinding.io/service-provider/
```

Worth adding the ref?

File: bitnami/postgresql/templates/secrets.yaml [hunk 2]
```diff
@@ -14,16 +22,62 @@ metadata:
...
-  replication-password: {{ include "common.secrets.passwords.manage" (dict "secret" (include "common.names.fullname" .) "key" "replication-password" "providedValues" (list "auth.replicationPassword") "context" $) }}
+  replication-password: {{ $replicationPassword | b64enc | quote }}
   {{- end }}
   # We don't auto-generate LDAP password when it's not provided as we do for other passwords
   {{- if and .Values.ldap.enabled (or .Values.ldap.bind_password .Values.ldap.bindpw) }}
-  ldap-password: {{ coalesce .Values.ldap.bind_password .Values.ldap.bindpw  | b64enc | quote }}
+  ldap-password: {{ $ldapPassword  | b64enc | quote }}
   {{- end }}
-{{- end -}}
+{{- if .Values.serviceBindings.enabled }}
```
Comment (mpermar) [inline bitnami/postgresql/templates/secrets.yaml:37]: Does it make sense to have this`if` condition outside of the global createSecret if. That way at least in some scenarios there would only be one secret.

Comment (javsalgar) [reply bitnami/postgresql/templates/secrets.yaml:37]: Not sure if I follow. There will be only one secret when `serviceBindings.enabled=false`, which will be the default secret. When this flag is set to true, we will have the default secret plus the serviceBindings one.

**Review by mpermar — APPROVED** [review]
Looks great.

**Pull request merged**